### PR TITLE
docs: Add KServe explainer tutorial

### PR DESCRIPTION
**Pull request by ruivieira**

### CODE REVIEW

File: docs/modules/ROOT/pages/component-kserve-explainer.adoc [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+== Features
+
+- **Explainability**: Integrated support for xref:local-explainers.adoc#lime[LIME] and xref:local-explainers.adoc#shap[SHAP] explanation methods to interpret model predictions via the `:explain` endpoint.
+
+== Deployment on KServe
+
+The TrustyAI explainer can be added to KServe `InferenceServices` and can be configured to use either xref:local-explainers.adoc#lime[LIME] or xref:local-explainers.adoc#shap[SHAP] explanation methods by modifying the YAML configuration.
+
+When deployed, Istio manages the routing of requests to the appropriate container. Calls to `/v1/models/model:predict` will be sent to the predictor container, while calls to `/v1/models/model:explain` will be sent to the explainer container. The payloads for both endpoints are the same, but the `:predict` endpoint returns the model's prediction, while the `:explain` endpoint returns an explanation of the prediction.
```
Comment (danielezonca) [inline docs/modules/ROOT/pages/component-kserve-explainer.adoc:15]: I would remove all Istio mentions from the document: from user PoV it is KServe that does it, Istio is an implementation detail. Moreover it is not the only option that is available ([Kourier should work too](https://kserve.github.io/website/latest/admin/serverless/kourier_networking/)).
Same for Istio logo: use KServe logo directly so we don't expose too many details

File: docs/modules/ROOT/examples/inference-service-explainer-lime.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+apiVersion: "serving.kserve.io/v1beta1"
+kind: "InferenceService"
+metadata:
+  name: "explainer-test-lime"
+  annotations:
+    sidecar.istio.io/inject: "true"
+    sidecar.istio.io/rewriteAppHTTPProbers: "true"
+    serving.knative.openshift.io/enablePassthrough: "true"
```
Comment (danielezonca) [inline docs/modules/ROOT/examples/inference-service-explainer-lime.yaml:8]: I think these annotations are necessary only for OpenShift Serverless + OpenShift Service Mesh not for upstream Istio (as far as I remember the default upstream is inject true) and Knative.

Btw this is a minor comment but you can mention in the document that has been tested using OpenShift + OpenShift Serverless + OpenShift Service Mesh and ODH

File: docs/modules/ROOT/pages/component-kserve-explainer.adoc (hunk 1 above)
Comment (ruivieira) [reply docs/modules/ROOT/pages/component-kserve-explainer.adoc]: Replaced Istio with KServe

File: docs/modules/ROOT/examples/inference-service-explainer-lime.yaml (hunk 2 above)
Comment (ruivieira) [reply docs/modules/ROOT/examples/inference-service-explainer-lime.yaml:8]: Removed annotations from the KServe/Kubernetes example

**Review by danielezonca — APPROVED** [review]
LGTM 👍

**Pull request merged**